packages/venia-concept/src/components/Checkout/addressForm.js

```diff
@@ -39,10 +41,29 @@ class AddressForm extends Component {
             postcode: string,
             region_code: string,
             street0: string,
-            telephone: string
+            telephone: string,
+            validation: string
         }),
         submit: func.isRequired,
-        submitting: bool
+        submitting: bool,
+        isAddressIncorrect: bool,
+        incorrectAddressMessage: string
+    };
+
+    //TODO: implement appropriate validation for the state field
+    validateState = value => {
+        return isString(value) && value.length > 1 ? null : invalidStateMessage;
+    };
```

Comment:
How does `checkout/asyncActions.js` know that the state is invalid? Can we re-use the same logic?

Reply:
As far as I understand this validator blocks user from submitting wrong value, so it will not come to asyncActions

Reply:
That's correct, I just meant maybe `asyncActions` and this file could use a common validator function